test/gtest/common/mem_buffer.cc

```diff
@@ -43,20 +43,41 @@
 #endif
 
 
+bool mem_buffer::is_cuda_supported()
+{
+#if HAVE_CUDA
+    int num_gpus;
+    cudaError_t cudaErr = cudaGetDeviceCount(&num_gpus);
+    return (cudaErr == cudaSuccess) || (num_gpus > 0);
```

Comment:
@yosefe Should this be `&&` instead of `||` ? Or should the tests run even if no GPUs are available on the system?

Reply:
right, this should be &&, shall i PR a fix or this can be part of your fix PR for HSA issue (same for ROCm)?

Reply:
Thanks, I will update my PR to address this